cloud/endagaweb/models.py

```diff
@@ -868,10 +884,73 @@ def set_subscriber_last_active(sender, instance=None, created=False,
         event.subscriber.last_active = event.date
         event.subscriber.save()
 
+    @staticmethod
+    def if_invalid_events(sender, instance=None, created=False, **kwargs):
+        # Check for any invalid event and make an entry
+        if not created:
+            return
+        event = instance
+        if event.kind in INVALID_EVENTS:
+            subscriber = Subscriber.objects.get(imsi=event.subscriber_imsi)
+            if SubscriberInvalidEvents.objects.filter(
+                    subscriber=event.subscriber).exists():
+                # Subscriber is blocked after 3 counts i.e there won't be UEs
+                # unless unblocked
+                subscriber_event = SubscriberInvalidEvents.objects.get(
+                    subscriber=event.subscriber)
+                # if it is a 3rd event in 24hr block the subscriber
+                negative_transactions_ids = subscriber_event.negative_transactions + [
+                    event.transaction_id]
+                subscriber_event.count = subscriber_event.count + 1
+                subscriber_event.event_time = event.date
+                subscriber_event.negative_transactions = negative_transactions_ids
+                subscriber_event.save()
+
+                max_transactions = event.subscriber.network.max_failure_transaction
+                if subscriber_event.count == max_transactions:
+                    subscriber.is_blocked = True
+                    subscriber.block_reason = 'Repeated %s within 24 hours ' % (
```

Comment:
do we need block reason? Can we just list out the negative_transactions in the UI?